src/DataBase/GoalDatabase.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using ePiggyWeb.DataBase.Models;
+using ePiggyWeb.DataManagement.Entries;
+using ePiggyWeb.DataManagement.Goals;
+using ePiggyWeb.Utilities;
+
+namespace ePiggyWeb.DataBase
+{
+    public static class GoalDatabase
+    {
+        public static int Create(IGoal goal, int userid)
+        {
+            using var db = new DatabaseContext();
+            var dbGoal = new GoalModel (goal, userid);
+            db.Add(dbGoal);
+            db.SaveChanges();
+            return dbGoal.Id;
+        }
+
+        public static bool CreateList(IGoalList goalList, int userid)
+        {
+            using var db = new DatabaseContext();
+            var dictionary = new Dictionary<IGoal, IGoalModel>();
+            foreach (var goal in goalList)
+            {
+                var dbGoal = new GoalModel(goal, userid);
+                dictionary.Add(goal, dbGoal);
+            }
+            // Setting all of the ID's to local Entries, just so this method remains usable both with local and only database usage
+            db.AddRange(dictionary.Values);
+            db.SaveChanges();
+            goalList.Clear();
+            foreach (var (key, value) in dictionary)
+            {
+                key.Id = value.Id;
+                goalList.Add(key);
+            }
+            return true;
+        }
+
+        public static bool Update(IGoal oldGoal, IGoal newGoal)
+        {
+            return Update(oldGoal.Id, oldGoal.UserId, newGoal);
+        }
+
+        public static bool Update(int id, int userId, IGoal newGoal)
+        {
+            using var db = new DatabaseContext();
+            var dbGoal = db.Goals.FirstOrDefault(x => x.Id == id && x.UserId == userId);
+            if (dbGoal == null)
+            {
+                ExceptionHandler.Log("Couldn't find goal id: " + id + " in database");
+                return false;
+            }
+            dbGoal.Edit(newGoal);
+            db.SaveChanges();
+            return true;
+        }
+
+        public static bool Delete(IGoal goal)
+        {
+            return Delete(goal.Id, goal.UserId);
+        }
+
+        public static bool Delete(int id, int userId)
+        {
+            return Delete(x => x.Id == id && x.UserId == userId);
+        }
+
+        public static bool Delete(Func<GoalModel, bool> filter)
```

Comment:
How is this stuff called? Delegates or smth? Or is this just simple passing function as parameter?

Reply:
Yeah, it's a delegate of Func type, takes in one parameter (GoalModel) and returns a result (bool). From line calling it with a lambda expression, but you could write in a different form, as long as it matches the Func<GoalModel,bool> type.
This link explains it more in depth: https://docs.microsoft.com/en-us/dotnet/api/system.func-2?view=netcore-3.1
In general I'm using this to pass a predicate to a method